pandas/core/series.py

```diff
@@ -2478,7 +2480,33 @@ def round(self, decimals=0, *args, **kwargs) -> Series:
 
         return result
 
-    def quantile(self, q=0.5, interpolation="linear"):
+    @overload
+    def quantile(
+        self, q: float = ..., interpolation: QuantileInterpolation = ...
+    ) -> float:
+        ...
+
+    @overload
+    def quantile(
+        self,
+        q: Sequence[float] | AnyArrayLike,
+        interpolation: QuantileInterpolation = ...,
+    ) -> Series:
+        ...
+
+    @overload
+    def quantile(
+        self,
+        q: float | Sequence[float] | AnyArrayLike = ...,
+        interpolation: QuantileInterpolation = ...,
+    ) -> float | Series:
+        ...
+
+    def quantile(
+        self,
+        q: float | Sequence[float] | AnyArrayLike = 0.5,
```

Comment:
Technically, I wouldn't need this overload to enable reportGeneralTypingIssues for `pandas/core/describe.py` - would only need `float | Sequence[float]`. In the absence of annotations, pyright assumes the default argument's type to be the annotation (in this case `float`).